### DISCUSSION TITLE
AGE-MOEA2 Capabilities

### DISCUSSION THREAD

**Discussion by muazhari** [anyoptimization/pymoo#618] [Q&A]
Does [AGE-MOEA2](https://github.com/anyoptimization/pymoo/blob/cb3f1169846b21d26a042aa71279df171b6e5dee/pymoo/algorithms/moo/age2.py) support constraints?

How can I know the capabilities of each algorithm outside from [this](https://pymoo.org/algorithms/list.html)?

And, does AGE-MOEA2 support mixed variable by supplying mixed variable mating, sampling, and duplicate elimination to it's parameters while sustaining it's default capabilities like supporting many objectives and constraints if true?

**Comment by blankjul** [comment]
Most algorithms don't support mixed variables. The variables are mostly implemented as proposed in the paper where more information are provided. 

(this link you are referring to provides an overview which so far is the correct source to see what is supported)

**Reply by muazhari** [reply]
![image](https://github.com/user-attachments/assets/c46ec0ba-f808-40ae-98d9-f4dd534055d9)

So, specifically, does [age2](https://github.com/anyoptimization/pymoo/blob/cb3f1169846b21d26a042aa71279df171b6e5dee/pymoo/algorithms/moo/age2.py) support constraints (the algorithm list link only provide information of age1, not age2) and [mixed variable](https://pymoo.org/customization/mixed.html) by modifying it as in the pict to the below implementation? (yes/no)

Code implementation:
```python
algorithm = AGEMOEA2(
    sampling=MixedVariableSampling(),
    mating=MixedVariableMating(eliminate_duplicates=MixedVariableDuplicateElimination()),
    eliminate_duplicates=MixedVariableDuplicateElimination(),
)
```
